recipes/stlab/all/conanfile.py

```diff
@@ -11,48 +11,196 @@ class Stlab(ConanFile):
+        # included in the OS.
+        return self.options.task_system == "libdispatch" and self.settings.os != "Macos"
+
+    def requirements(self):
+        if self._use_boost():
+            self.requires("boost/1.75.0")
+
+        if self._requires_libdispatch():
+            self.requires("libdispatch/5.3.2")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = "libraries-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
-    def configure(self):
+    def _configure_boost(self):
+        self.options["boost"].shared = True
+
+        self.options["boost"].without_atomic = True
+        self.options["boost"].without_chrono = True
+        self.options["boost"].without_container = True
+        self.options["boost"].without_context = True
+        self.options["boost"].without_contract = True
+        self.options["boost"].without_coroutine = True
+        self.options["boost"].without_date_time = True
+        self.options["boost"].without_exception = True
+        self.options["boost"].without_fiber = True
+        self.options["boost"].without_filesystem = True
+        self.options["boost"].without_graph = True
+        self.options["boost"].without_graph_parallel = True
+        self.options["boost"].without_iostreams = True
+        self.options["boost"].without_json = True
+        self.options["boost"].without_locale = True
+        self.options["boost"].without_log = True
+        self.options["boost"].without_math = True
+        self.options["boost"].without_mpi = True
+        self.options["boost"].without_nowide = True
+        self.options["boost"].without_program_options = True
+        self.options["boost"].without_python = True
+        self.options["boost"].without_random = True
+        self.options["boost"].without_regex = True
+        self.options["boost"].without_serialization = True
+        self.options["boost"].without_stacktrace = True
+        self.options["boost"].without_system = True
+        self.options["boost"].without_test = True
+        self.options["boost"].without_timer = True
+        self.options["boost"].without_thread = True
+        self.options["boost"].without_type_erasure = True
+        self.options["boost"].without_wave = True
+
+    def _fix_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+
+        #
+        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 is not complete.
+        #
+        self.output.info("Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead.")
+        self.options.boost_optional = True
+        self.options.boost_variant = True
```

Comment:
overriding options silently is really not ideal. it better to just raise the minimum and leave a not (though it's a very common problem here)